solr/core/src/java/org/apache/solr/search/facet/UnInvertedField.java

```diff
@@ -682,15 +682,16 @@ protected static void rethrowAsSolrException(String field, Throwable e) {
   }
 
   // Returns null if not already populated
-  @SuppressWarnings({"rawtypes", "unchecked"})
   public static UnInvertedField checkUnInvertedField(String field, SolrIndexSearcher searcher) throws IOException {
-    SolrCache cache = searcher.getFieldValueCache();
+    SolrCache<String, UnInvertedField> cache = searcher.getFieldValueCache();
     if (cache == null) {
       return null;
     }
     Object uif = cache.get(field);  // cache is already synchronized, so no extra sync needed
     // placeholder is an implementation detail, keep it hidden and return null if that is what we got
     return uif==uifPlaceholder || !(uif instanceof UnInvertedField)? null : (UnInvertedField) uif;
+    // TODO: SolrCache is not used safely in other places, but this might be simpligfied to:
+    //  return uif==uifPlaceholder ? null : uif;
```

Comment:
Could you share more re: what other places do not safely use SolrCache? I agree with the "this might be simplified" bit, from looking at the code here only.

```suggestion
    UnInvertedField uif = cache.get(field);  // cache is already synchronized, so no extra sync needed
    // placeholder is an implementation detail, keep it hidden and return null if that is what we got
    return uif==uifPlaceholder || uif==null ? null : uif;
```

And I'd like to note that the `|| uif==null` is redundant but without it the "return null if that is what we got" part of the comment would be less clear, though we could somehow change the comment too of course.

Reply:
There is a lot of raw usage of SolrCache around the code, mainly around SolrIndexSearcher.getCache, so I wasn't confident enough in all the code paths not hiding any other misuse to get rid of the instances safety check.